app/controllers/volunteers_controller.rb

```diff
@@ -20,32 +20,38 @@ def new
           redirect_to volunteer_path(current_user.volunteer.id)
     end
   end
-  
+
   def show
     @volunteer = current_user.volunteer
   end
-  
+
   def create
     # @repair = Repair.all
     @volunteer = Volunteer.new(volunteer_params)
     @volunteer.user_id = current_user.id
     if @volunteer.save
-          redirect_to volunteer_path(@volunteer.id)
+      flash[:success] = "Volunteer created!"
+      redirect_to volunteer_path(@volunteer)
     else
       render 'new'
     end
   end
 
   def edit
     # @repair = Repair.all
-    @volunteer = Volunteer.find(current_user.id)
+    @volunteer = Volunteer.find(params[:id])
+    @volunteer.user_id = current_user.id
   end
 
   def update
-      volunteer = Volunteer.find(params[:id])
-      volunteer.update!(volunteer_params)
-      volunteer.user_id = current_user.id  #put this seperate because doesn't seem to work included as strong params
-      redirect_to volunteer_path(volunteer.id)
+      @volunteer = Volunteer.find(params[:id])
```

Comment:
Same problem.  If the id posted is not the current_user.id, the profile being updated is going to be someone else's.  This is a security exposure.